index.js

```diff
@@ -1,18 +1,27 @@
 'use strict';
 const colorConvert = require('color-convert');
 
-const wrapAnsi16 = (fn, offset) => function () {
+const wrapAnsi16 = (key, fn, offset) => function () {
+	if (key === 'keyword' && colorConvert.keyword.rgb.apply(colorConvert, arguments) === undefined) {
+		throw new Error(`Invalid arguments provided to ${key}: ${Array.from(arguments)}`);
+	}
 	const code = fn.apply(colorConvert, arguments);
 	return `\u001B[${code + offset}m`;
 };
 
-const wrapAnsi256 = (fn, offset) => function () {
+const wrapAnsi256 = (key, fn, offset) => function () {
+	if (key === 'keyword' && colorConvert.keyword.rgb.apply(colorConvert, arguments) === undefined) {
+		throw new Error(`Invalid arguments provided to ${key}: ${Array.from(arguments)}`);
+	}
 	const code = fn.apply(colorConvert, arguments);
 	return `\u001B[${38 + offset};5;${code}m`;
 };
 
-const wrapAnsi16m = (fn, offset) => function () {
+const wrapAnsi16m = (key, fn, offset) => function () {
 	const rgb = fn.apply(colorConvert, arguments);
+	if (rgb === null || rgb === undefined) {
+		throw new Error(`Invalid arguments provided to ${key}: ${Array.from(arguments)}`);
+	}
```

Comment:
Could you persist a value using `let rgb;` instead? e.g.

```javascript
let rgb;
if (key === 'keyword') {
    rgb = colorConvert.keyword.rgb.apply(...);
    if (rgb === undefined) {
        ...
    }
} else {
    rgb = fn.apply(colorConvert, arguments);
}

```

Reply:
It's a little more verbose but it prevents evaluating the keyword function multiple times.

Reply:
@Qix- 
In `wrapAnsi16m`, it's only being calculated once.

In `wrapAnsi256` and `wrapAnsi`, I don't see any other way to prevent duplicate calls other than this. Is this what you had in mind? At this point I'm having to duplicate the logic that's buried in `color-convert`, which I'm fine with but I feel that needs your blessing before I plow ahead.
```
const wrapAnsi256 = (key, fn, offset) => function () {
  let code;
  if (key === 'keyword'){
    const rgb = colorConvert.keyword.rgb.apply(colorConvert, arguments)
    if (rgb === undefined) {
      throw new Error(`Invalid arguments provided to ${key}: ${Array.from(arguments)}`);
    }
    code = colorConvert.rgb.ansi256(rgb)
  } else {
    code = fn.apply(colorConvert, arguments)
  }
  return `\u001B[${38 + offset};5;${code}m`
}
```

At this point, it would probably be clearer to explicitly define the keyword mapper on the ansi and ansi256 objects, much like the rgb mapper is explicitly defined on the ansi16m object done at line 113/122

Reply:
@Qix- Here's a different implementation with the explicitly defined mappers. It certainly makes the wrapAnsi methods simpler, and I'm a fan of simple code. 

I'm not quite sure the normal flow for alternative solutions to PRs, and I didn't feel like you'd want me to submit a second PR at this time, so here's my changes in compare view from my fork: https://github.com/chalk/ansi-styles/compare/master...danthegoodman:keyword_missing_error_2

Reply:
@danthegoodman ^ that diff looks much better to me :)

Reply:
@danthegoodman Just push another commit with the changes ;)